docs/howto/move-partitioned-database.md

```diff
@@ -0,0 +1,155 @@
+# Moving a PostgreSQL sharded database
+
+For large scale deployments of CommCare HQ the database tables that contain
+case and form (and related) data can be partitioned between multiple PostgreSQL databases.
+
+To ease the burden of scaling it is common to create as many databases as necessary
+up front since splitting databases is much more difficult than moving databases. In
+this case a deployment of CommCare HQ may start with a single database server
+that contains all the databases. As the project scales up more database servers
+can be added and the databases can be moved to spread the load.
+
+This document describes the process required to move a database from one PostgreSQL
+instance to another.
+
+For the purposes of this document we'll assume that we have two database machines, *pg1*
+and *pg2*. *pg1* has two partitioned databases and *pg2* has none:
+
+*pg1* databases:
+
+* *partition1*
+* *partition2*
+
+*pg2* is a newly deployed server and we want to move *partition2* onto *pg2*.
+
+## Assumptions
+
+* *pg2* has been added to the Ansible inventory
+* *pg2* has had a full stack Ansible deploy
+* *pg1* has a replication slot available
+
+## Process Overview
+
+1. Setup *pg2* node as a standby node of *pg1*
+3. Promote *pg2* to a master node
+4. Update the configuration so that requests for *partition2* go to *pg2* instead
+of *pg1*.
+
+## Process detail
+
+### 1. Setup *pg2* as a standby node of *pg1*
+This step does not require downtime and can be done at any stage prior to the
+downtime.
+
+```
+commcare-cloud <env> ansible-playbook setup_pg_standby.yml -e standby=pg2 -e hot_standby_master=pg1 -e replication_slot=[replication slot name]
```

Comment:
It's the same command but for normal standbys you don't have to supply the `hot_standby_master` or the `replication_slot` params since they are in the inventory.